.github/workflows/check.yaml

```diff
@@ -1,140 +1,180 @@
-             # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-masking-a-generated-output-within-a-single-job
-             echo ${{ secrets.OPENSTACK_AUTH_ENV }} \
-                | base64 -d \
-                | grep -E 'OS_AUTH_URL|OS_USERNAME|OS_PASSWORD|OS_PROJECT_NAME|OS_PROJECT_ID' \
-                | awk -F'=' '{print $2}' \
-                | xargs -I {} echo "::add-mask::{}"
-             echo ${{ secrets.OPENSTACK_AUTH_ENV }} | base64 -d >> "$GITHUB_ENV"
-           else
-             echo "Skip because missing openstack-auth-env secrets"
-           fi
-
-      - name: append to zaza yaml
-        run: |
-           cat >> $HOME/.zaza.yaml <<EOF
-           ---
-           model_settings:
-             image-stream: released
-             update-status-hook-interval: 30s
-           region: prodstack6
-           cloud: builder-cloud
-           credential: builder-cloud-cred
-           EOF
-
-      - name: Configure latest lxd (to build the charm in)
-        uses: canonical/setup-lxd@2aa6f7caf7d1484298a64192f7f63a6684e648a4  # latest main at time of writing
+      - name: Setup Juju environment
+        uses: charmed-kubernetes/actions-operator@main
         with:
-          channel: latest/stable
+          provider: "lxd"
+          juju-channel: ${{ matrix.juju-channel }}
+          charmcraft-channel: "2.x/stable"
 
-      - name: install external controller required packages
+        # This is used by zaza in the functional tests for non-amd64 architectures (if applicable)
+      - name: Set zaza juju model constraints for architecture
         run: |
-           sudo snap install charmcraft --channel 3.x/stable --classic
-           sudo snap install juju --channel 3.4/stable --classic
-           sudo --preserve-env=http_proxy,https_proxy,no_proxy pip3 install tox
-           juju switch soleng-ci-ctrl-34
-           # Workaround to avoid missing cookies bug
-           # https://github.com/search?q=repo%3Ajuju%2Fpython-libjuju%20cookies_for_controller&type=code
-           juju show-controller soleng-ci-ctrl-34 &>/dev/null
-
-      - name: Show juju information
+          if [ "$(uname -m)" = "aarch64" ]; then
+            echo "TEST_MODEL_CONSTRAINTS=arch=arm64" >> "$GITHUB_ENV"
+          fi
+
+      - name: Enabled nested LXD containers
         run: |
-          juju version
-          juju controllers | grep Version -A 1 | awk '{print $9}'
+          # create new lxd project and configure nesting & privileged
+          lxc profile show default > default-lxc-profile.yaml
+          lxc project create nested-lxd
+          lxc profile edit --project nested-lxd default < default-lxc-profile.yaml
+          lxc profile set --project nested-lxd default security.nesting=true
+          lxc profile set --project nested-lxd default security.privileged=true
+          # set default project to be nested-lxd project
+          juju model-default project=nested-lxd
```

Comment:
I don't think we need this, because this is for nested containers. But we want nested VMs.